pkg/core/server/registry_bridge.go

```diff
@@ -341,6 +347,10 @@ func (s *Server) isValidRegisterNodeTx(ctx context.Context, tx *core_proto.Signe
 	onChainBlockNumber := info.BlockNumber.String()
 	onChainEndpoint := info.Endpoint
 
+	if err := s.isDuplicateDelegateOwnerWallet(onChainOwnerWallet); err != nil {
```

Comment:
since we're caching the nodes, can we remove the synchronous calls to etheruem above and just fetch it from the list we already have?